### PR TITLE
[Enhance] Support arbitrary seq_len in Linear Attention

### PR DESCRIPTION
**Pull request by Rachmanino**
1. Support arbitrary seqlen in Linear Attention via padding
2. Add `utils.py` to provide utility functions
3. Fix a typo in `mha.py`

### CODE REVIEW

File: top/kernel/linear_attention/linear_attn.py [hunk 1]
```diff
@@ -231,16 +219,22 @@ def __init__(self,
...
         self.block_K = block_K
         self.block_V = block_V
         self.chunk_size = chunk_size
 
         self.attention = fused_chunk_linear_attention
 
     def forward(self, q, k, v, scale=None):  # Layout: [B, S, H, D]
-        return self.attention(q, k, v, scale, self.dtype, self.block_K, self.block_V,
+        if self.seq_len % self.chunk_size != 0:
+            q, k, v = map(lambda x: zero_pad(x, self.chunk_size, 1), (q, k, v))
```
[the remark this answers is not in the document]
Comment (Rachmanino) [reply top/kernel/linear_attention/linear_attn.py:228]: Actually `self.seq_len` should be set to equal the actual sequence length of the input tensors?

File: top/kernel/mha.py [hunk 2]
```diff
@@ -313,22 +313,10 @@ def ref_program(cls, q, k, v, causal):
...
-                        requires_grad=True)
-        v = torch.randn((self.batch_size, self.seq_len, self.num_heads, self.head_dim),
-                        device=self.device,
-                        dtype=self.dtype,
-                        requires_grad=True)
-        do = torch.randn((self.batch_size, self.seq_len, self.num_heads, self.head_dim),
-                         device=self.device,
-                         dtype=self.dtype)
-        return q, k, v, do
+                        requires_grad=True) for _ in range(4))
```
[the remark this answers is not in the document]
Comment (chengyupku) [reply top/kernel/mha.py]: I think `do` tensor should not require gradients?

**Comment by Rachmanino** [comment]
Fixed. Please take a look, thx. @chengyupku
